packages/tupaia-web/src/features/EnlargedDashboardItem/utils/useExportDashboardItem.tsx

```diff
@@ -0,0 +1,185 @@
+  Entity,
+  EntityCode,
+  ProjectCode,
+} from '../../../types';
+import { gaEvent } from '../../../utils';
+import {
+  ACTION_TYPES,
+  EXPORT_FORMATS,
+  ExportContext,
+  ExportDispatchContext,
+} from './ExportContext';
+import { useExportToExcel } from '../../../api/queries';
+
+interface ExportToExcelParams {
+  projectCode?: ProjectCode;
+  dashboardCode?: Dashboard['code'];
+  entityCode?: EntityCode;
+  config?: DashboardItemConfig;
+  report?: DashboardItemReport;
+  filename?: string;
+  currentDashboardItem?: DashboardItem;
+}
+// generate the params for the useExportToExcel hook
+export const getExportToExcelParams = ({
+  projectCode,
+  dashboardCode,
+  entityCode,
+  config,
+  report,
+  currentDashboardItem,
+}: ExportToExcelParams) => {
+  const { exportConfig } = config || {};
+  const dataElementHeader = exportConfig?.dataElementHeader;
+  const { legacy, code: itemCode } = currentDashboardItem || {};
+  const { startDate, endDate } = report || ({} as DashboardItemReport);
+
+  let newStartDate = moment.isMoment(startDate) ? startDate.utc().toISOString() : startDate;
+  let newEndDate = moment.isMoment(endDate) ? endDate.utc().toISOString() : endDate;
+  if (!startDate && !endDate) {
+    const { startDate: viewStartDate, endDate: viewEndDate } = report || {};
+    newStartDate = viewStartDate;
+    newEndDate = viewEndDate;
+  }
+
+  return {
+    projectCode,
+    dashboardCode,
+    entityCode,
+    itemCode,
+    dataElementHeader,
+    startDate: newStartDate,
+    endDate: newEndDate,
+    legacy,
+  };
+};
+
+// This is a utility hook that is used in the ExportDashboardItem component, that handles all the exporting of single dashboard items
+export const useExportDashboardItem = (
+  activeDashboard?: Dashboard | null,
+  currentDashboardItem?: DashboardItem,
+  report?: DashboardItemReport,
+  entityName?: Entity['name'],
+  exportRef?: RefObject<HTMLElement>,
+) => {
+  const { projectCode, entityCode } = useParams();
+  const {
+    exportFormat,
+    exportWithLabels,
+    exportWithTable,
+    exportWithTableDisabled,
+    isExporting,
+    exportError,
+    isExportMode,
+  } = useContext(ExportContext);
+  const dispatch = useContext(ExportDispatchContext)!;
+  const { config } = currentDashboardItem || ({} as DashboardItem);
+  const { type, presentationOptions, name } = config || ({} as DashboardItemConfig);
+  const exportTitle = `${name}, ${entityName}`;
+
+  const { doExport } = useChartDataExport(
+    {
+      ...report,
+      ...config,
+      presentationOptions: {
+        ...(presentationOptions || {}),
+        exportWithLabels,
+        exportWithTable,
+        exportWithTableDisabled,
+      },
+    } as ViewContent,
+    exportTitle,
+  );
+
+  // use the useExportToExcel hook to export the matrix dashboard item. We disable the fetch initially so that we can only do the query when we want to export
+  const { error: excelError, refetch: callExportToExcel, isFetching } = useExportToExcel(
```

Comment:
I think useExportToExcel should be a [mutation](https://tanstack.com/query/v3/docs/react/guides/mutations) since the api request gets triggered by a user click (I think it counts as a server-side-effect in react-query vernacular). Them we wouldn't have to use the refetch function to make the request which doesn't quite seem right.